modules/standard/Maps.chpl

```diff
@@ -0,0 +1,291 @@
+    */
+    inline proc const size {
+
+    }
+
+    /*
+      Returns `true` if this map contains zero keys.
+
+      :returns: `true` if this map is empty.
+      :rtype: `bool`
+    */
+    inline proc const isEmpty(): bool {
+
+    }
+
+    /*
+      Returns `true` if the given key is a member of this map, and `false`
+      otherwise.
+
+      :arg k: The key to test for membership.
+      :type k: keyType
+
+      :returns: Whether or not the given key is a member of this map.
+      :rtype: `bool`
+    */
+    proc const contains(const ref k: keyType): bool {
+
+    }
+
+    /*
+      Updates this map with the contents of the other, overwritting the values
+      for already-existing keys.
+
+      :arg m: The other map
+      :type m: map(keyType, valType)
+    */
+    proc update(const ref m: zap(keyType,valType)) {
+
+    }
+
+    /*
+      Get a reference to the value mapped to the given key.
+
+      :arg k: The key to access
+      :type k: keyType
+
+      .. warning::
+
+        Use of the `this` method with a non-existent key (while bounds checking
+        is on) will cause the currently running program to halt.
+
+      :returns: Reference to the value mapped to the given key.
+    */
+    proc const this(k: keyType) ref {
+
+    }
+
+    /*
+      Iterates over the keys of this map.
+
+      :yields: A reference to one of the keys contained in this map.
+    */
+    iter these() const ref {
+
+    }
+
+    /*
+      Iterates over the key-value pairs of this map.
+
+      :yields: A tuple of references to one of the key-value pairs contained in
+               this map.
+    */
+    iter items() const ref {
+
+    }
+
+    /*
+      Iterates over the values of this map.
+
+      :yields: A reference to one of the values contained in this map.
+    */
+    iter values() ref {
+
+    }
+
+    /*
+      Writes the contents of this map to a channel. The format looks like:
+
+        .. code-block:: chapel
+    
+           {k1: v1, k2: v2, .... , kn: vn}
+
+      :arg ch: A channel to write to.
+    */
+    proc writeThis(ch: channel) {
+
+    }
+
+    /*
+      Adds a key-value pair to the map. Different than doing `myMap[key] =
+      value`, this method returns `false` if the key already exists in the map.
+
+     :arg k: The key to add to the map
+     :type k: keyType
+
+     :arg v: The value that maps to ``k``
+     :type k: valueType
+
+     :returns: `true` if `k` was not in the map and added with value `v`.
+               `false` otherwise.
+     :rtype: bool
+    */
+    proc add(k: keyType, v: valType): bool {
+
+    }
+
+    /*
+      Removes a key-value pair to the map, withe the given key.
+      
+     :arg k: The key to remove from the map
+     :type k: keyType
+
+     :returns: `false` if `k` was not in the map.  `true` if it was and removed.
+     :rtype: bool
+    */
+    proc remove(k: keyType): bool {
+
+    }
+
+    /*
+      Returns a new 0-based array containing a copy of key-value pairs as
```

Comment:
I'm not quite sure why this is a 0-based array. Should we update `list` and `set` to match? I'd also like a line mentioning that the keys are not in any particular order.

Reply:
I can make this match list and set by returning a 1-based array. I just thought it is important that this is stated in the doc. I will update this.